include/parsers/qasm3_parser/Scanner.hpp

```diff
@@ -0,0 +1,72 @@
+#pragma once
+
+#include "Token.hpp"
+
+#include <fstream>
+#include <iostream>
+#include <map>
+#include <sstream>
+#include <stack>
+
+namespace qasm3 {
+class Scanner {
+private:
+  std::istream& is;
```

Comment:
Maybe prefer to make this either a plain `std::istream*` or a `std::shared_ptr`. Reference members prevent copy construction etc.